File: benchmarks/Java/safety/simple/even_list_nondeterm.java
class List extends Object {
    List() { super(); }
    Boolean iseven(){ assert false; }
}
class Nil extends List {
    Nil() { super(); }
    Boolean iseven() { return true; }
}
class Cons extends List {
    Int hd;  List tl;
    Cons(Int hd, List tl) { super(); this.hd = hd; this.tl = tl; }
    Boolean iseven() { return !this.tl.iseven(); }
}
class Main extends Object {
    Main() { super(); }
    List mk_evenlist() {
        return if_ then new Nil() else new Cons(1, new Cons(2, this.mk_evenlist()));
    }
}
main(){
    assert new Main().mk_evenlist().iseven();
}

File: benchmarks/OCaml/safety/objects/even_list_nondeterm.ml
class virtual list = object
  method virtual iseven : bool
end

class nil = object
  inherit list
  method iseven = true
end

class cons (hd:int) (tl:list)  = object
  inherit list
  method iseven = not (tl#iseven)
end

class main = object (s)
  method mk_evenlist =
    if Random.bool () then new nil else new cons 1 (new cons 2 (s#mk_evenlist))
end

let main () =
    assert (new main#mk_evenlist#iseven)
